apps/motogp/motogp.star

```diff
@@ -0,0 +1,247 @@
+        expanded = True,
+        main_align = "space_between",
+        cross_align = "end",
+        children = [
+            render.Box(
+                width = 64,
+                height = 8,
+                color = RED_LOGO_COLOR,
+                child = render.Text(content = title, color = WHITE_COLOR, font = LOGO_FONT),
+            ),
+        ],
+    )
+
+def render_text_row(text, scroll = False):
+    content_cols = padded_text_row(text)
+    if scroll:
+        content_cols = [
+            render.Marquee(
+                offset_start = 32,
+                offset_end = 32,
+                align = "center",
+                width = 64,
+                child = render.Row(
+                    main_align = "start",
+                    children = content_cols,
+                ),
+            ),
+        ]
+    return render.Padding(
+        pad = (0, 0, 0, 0),
+        child = render.Box(
+            height = 6,
+            child = render.Row(
+                main_align = "center",
+                expanded = True,
+                children = content_cols,
+            ),
+        ),
+    )
+
+def render_rider_row(full_name, points):
+    ns = full_name.split(" ")
+    name = "{}. {}".format(ns[0][0], " ".join(ns[1:]))
+    name_cols = padded_text_row(name)
+    return render.Padding(
+        pad = (0, 1, 0, 0),
+        child = render.Row(
+            children = [
+                render.Marquee(
+                    offset_start = 64,
+                    offset_end = 64,
+                    width = 46,
+                    child = render.Row(children = name_cols),
+                ),
+                padded_text_row("-")[0],
+                render.Text(content = "{}".format(points), color = "#FFCE00", font = CAPS_FONT),
+            ],
+        ),
+    )
+
+def padded_text_row(text, color = WHITE_COLOR, font = CAPS_FONT):
+    content_cols = []
+    for s in text.split(" "):
+        content_cols.append(
+            render.Padding(
+                pad = (1, 0, 1, 0),
+                child = render.Text(content = s, color = color, font = font),
+            ),
+        )
+    return content_cols
+
+def fetch_calendar(year):
+    url = CALENDAR_URL.format(year = year)
+
+    # schedule can be cached for a day
+    data = get_cachable_data(url, 24 * ONE_HOUR)
+    return data
+
+def fetch_standings(year):
+    season_id = SEASONS_YEAR_ID[int(year)]
+
+    # categories list changes only once per year, a 3 day cache is good enough
+    categories = get_cachable_data(CATEGORIES_URL.format(season_id = season_id), 3 * ONE_DAY)
+    category_id = None
+    for category in categories:
+        if "motogp" in category["name"].lower():
+            category_id = category["id"]
+            break
+    url = STANDINGS_URL.format(season_id = season_id, category_id = category_id)
+    data = get_cachable_data(url, 1 * ONE_HOUR)
+    return data
+
+def get_cachable_data(url, timeout):
+    cached_data = cache.get(url)
```

Comment:
Would be great if you could switch this over to use the http module's built in caching instead? Just pass `ttl_seconds` to `http.get()`.

Reply:
Done